Visualizer/plotgame.py

```diff
@@ -41,36 +54,36 @@ def main():
             # if we press m1 we check what block the cursor is on and mark it as an obstacle
             if pg.mouse.get_pressed()[0]:
                 mouse_pos = pg.mouse.get_pos()
-                for row in blocks:
+                for row in npblocks:
                     for block in row:
-                        if block.body.collidepoint(mouse_pos):
+                        if block.rect.collidepoint(mouse_pos):
                             block.mark_obstacle()
 
             if event.type == pg.KEYDOWN:
```

Comment:
It would be a bad idea to have an input class or something of the sorts that takes in the input and stores it. for remapping purposes, now out of scope but more a thought.

Reply:
You meant it's a good idea right?

Reply:
indeed, I did an edit.